### PR TITLE
[gen] Make CRUD a simple enum

### PR DESCRIPTION
**Pull request by c-harding**
Stacks on #161, using native enums so we can use sorting on collections of endpoints

### CODE REVIEW

File: src/main/scala/temple/generate/CRUD.scala [hunk 1]
```diff
@@ -1,13 +1,6 @@
...
-object CRUD {
-  case object ReadAll extends CRUD("List")
-  case object Create  extends CRUD("Create")
-  case object Read    extends CRUD("Read")
-  case object Update  extends CRUD("Update")
-  case object Delete  extends CRUD("Delete")
-
-  val values = Seq(ReadAll, Create, Read, Update, Delete)
+object CRUD extends Enumeration {
+  type CRUD = Value
```
Comment (jaylees14) [inline src/main/scala/temple/generate/CRUD.scala:4]: Where for art thou value defined?

**Review by jaylees14 — APPROVED** [review]
LGTM, indifferent if we change or not, agree does read a little nicer

File: src/main/scala/temple/builder/MetricsBuilder.scala [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package temple.builder
 
-import temple.generate.CRUD
+import temple.generate.CRUD.CRUD
```
Comment (willjvsmith) [inline src/main/scala/temple/builder/MetricsBuilder.scala:3]: nice

**Review by willjvsmith — COMMENTED** [review]
Some of the functions in the Go generators refer to `ReadAll` in their names, probs want changing too

File: src/main/scala/temple/generate/CRUD.scala (hunk 1 above)
Comment (c-harding) [reply src/main/scala/temple/generate/CRUD.scala:4]: @jaylees14 it’s a field of the Enumeration class: https://www.scala-lang.org/api/current/scala/Enumeration.html

**Comment by willjvsmith** [comment]
Will approve once updated and conflicts resolved

**Review by willjvsmith — APPROVED** [review]

**Pull request merged**